packages/py-api/py_api/controllers/uploader_controller.py

```diff
@@ -69,3 +75,20 @@ def dump_objects(cls) -> Dict[str, Any]:
             )
 
         return {'objects': objects}
+
+    @classmethod
+    def delete_object_by_filename(cls, filename: str) -> Dict[str, Any]:
+        # Check whether the image exists
+        try:
+            cls._s3_client.head_object(
+                Bucket=cls._AWS_BUCKET_NAME, Key=filename,
+            )
+        except cls._s3_client.exceptions.ClientError as e:
+            if e.response['Error']['Code'] == '404':
+                raise HTTPException(
+                    status_code=404, detail=f"Object '{filename}' not found in the storage",
+                )
+
+        # If the object exists, proceed with the deletion
+        cls._s3_client.delete_object(Bucket=cls._AWS_BUCKET_NAME, Key=filename)
```

Comment:
Does this throw an exception when we try to delete an object which doesn't exist? If that's the case, remove the code above with the `head_object` implementation and wrap the `delete_object` invocation in a try catch clause. This should be done either way so that a proper response can be delivered to the user.